File: Java/Examples4/Sphere.java
public class Sphere implements IThreeDimensionalObject {

    public boolean isInside(double x, double y, double z) {
	if ( (x*x + y*y + z*z) <= 1.0 )
	    return true;
	else 
	    return false;
    }

    public String getName() {
	return "球";
    }

    public double getAnalyticalSolution() {
	return Math.PI*4.0/3.0;
    }

}

File: Python/Examples2/sphere.py
from i3d_object import IThreeDimensionalObject
import math

class Sphere(IThreeDimensionalObject):
    
    def is_inside(self, x, y, z):
        if (x**2 + y**2 + z**2) <= 1.0:
            return True
        else:
            return False
            
    @property
    def name(self):
        return '球'  
        
    @property    
    def analytical_solution(self):
        return math.pi*4.0/3.0
